packages/dialog/index.tsx

```diff
@@ -0,0 +1,322 @@
+    this.foundation = new MDCDialogFoundation(this.adapter);
+    this.foundation.init();
+    this.initializeFocusTrap();
+
+    if (open) {
+      this.foundation.open();
+    }
+    if (!autoStackButtons) {
+      this.foundation.setAutoStackButtons(autoStackButtons);
+    }
+
+    if (escapeKeyAction) {
+      this.foundation.setEscapeKeyAction(escapeKeyAction);
+    }
+
+    if (scrimClickAction) {
+      this.foundation.setScrimClickAction(scrimClickAction);
+    }
+  }
+
+  componentWillUnmount() {
+    this.foundation.destroy();
+  }
+
+  componentDidUpdate(prevProps: DialogProps<T>) {
+    const {open, autoStackButtons, escapeKeyAction, scrimClickAction} = this.props;
+
+    if (prevProps.autoStackButtons !== autoStackButtons) {
+      this.foundation.setAutoStackButtons(autoStackButtons);
+    }
+
+    if (prevProps.escapeKeyAction !== escapeKeyAction) {
+      this.foundation.setEscapeKeyAction(escapeKeyAction);
+    }
+
+    if (prevProps.scrimClickAction !== scrimClickAction) {
+      this.foundation.setScrimClickAction(scrimClickAction);
+    }
+
+    if (prevProps.open !== open) {
+      return open ? this.foundation.open() : this.foundation.close();
+    }
+  }
+
+  get classes(): string {
+    const {classList} = this.state;
+    const {className} = this.props;
+    return classnames(cssClasses.BASE, Array.from(classList), className);
+  }
+
+  get buttons_(): HTMLButtonElement[] | null {
+    return this.dialogElement.current &&
+       [].slice.call(this.dialogElement.current.getElementsByClassName(cssClasses.BUTTON));
+  }
+
+  get content_(): HTMLElement | null {
+    return this.dialogElement.current &&
+        this.dialogElement.current.querySelector(`.${cssClasses.CONTENT}`);
+  }
+
+  get defaultButton_(): HTMLButtonElement | null {
+    return this.dialogElement.current &&
+        this.dialogElement.current.querySelector(`.${cssClasses.DEFAULT_BUTTON}`);
+  }
+
+  private initializeFocusTrap = () => {
+    this.focusTrap = util.createFocusTrapInstance(this.dialogElement.current);
+  }
+
+  get adapter(): Partial<MDCDialogAdapter> {
+    const strings = MDCDialogFoundation.strings;
+    const {closest, matches} = ponyfill;
+    const {isScrollable, areTopsMisaligned} = util;
+    return {
+      addClass: (className: string) => {
+        const {classList} = this.state;
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className: string) => {
+        const {classList} = this.state;
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
+      addBodyClass: (className: string) => document.body.classList.add(className),
+      removeBodyClass: (className: string) => document.body.classList.remove(className),
+      eventTargetMatches: (target: HTMLElement, selector: string) => matches(target, selector),
```

Comment:
This and `getActionFromEvent` use the `dom` package, which can't be triggered from SSR initial load. It seems like these adapter methods are only called from `handleInteraction` (not initial load) - so I think we are ok.

Just wanted you to be aware of SSR issues (in case you see some issue). But I think we are ok.